src/main/java/seedu/address/ui/ActivityCard.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.ui;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import seedu.address.model.activity.Activity;
+
+/**
+ * An UI component that displays information of an {@code Activity}.
+ */
+public class ActivityCard extends UiPart<Region> {
+
+    private static final String FXML = "ActivityCard.fxml";
+
+    /**
+     * Note: Certain keywords such as "location" and "resources" are reserved keywords in JavaFX.
+     * As a consequence, UI elements' variable names cannot be set to such keywords
+     * or an exception will be thrown by JavaFX during runtime.
+     *
+     * @see <a href="https://github.com/se-edu/addressbook-level4/issues/336">The issue on AddressBook level 4</a>
+     */
+
+    public final Activity activity;
+
+    @FXML
+    private HBox cardPane;
+    @FXML
+    private Label index;
+    @FXML
+    private Label id;
+    @FXML
+    private Label title;
+    @FXML
+    private Label participantCount;
+
+    public ActivityCard(Activity activity, int displayedIndex) {
+        super(FXML);
+        this.activity = activity;
+        id.setText("ID: " + activity.getPrimaryKey());
+        index.setText("#" + displayedIndex);
+        title.setText(activity.getTitle().title);
+        int numParticipants = activity.getParticipantIds().size() + 1;
```

Comment:
I just noticed this, why is the number of participants 1 more than the actual number in the activity?